bb/bb.go

```diff
@@ -300,43 +307,43 @@ func oneFile(dir, s string, fset *token.FileSet, f *ast.File) error {
 		// Write the file to interface to the command package.
 		t := template.Must(template.New("cmdFunc").Parse(cmdFunc))
 		var b bytes.Buffer
-		if err := t.Execute(&b, config); err != nil {
+		if err := t.Execute(&b, c); err != nil {
 			log.Fatalf("spec %v: %v\n", cmdFunc, err)
 		}
 		fullCode, err := imports.Process("commandline", []byte(b.Bytes()), &opts)
 		if err != nil {
 			log.Fatalf("bad parse: '%v': %v", out, err)
 		}
-		if err := ioutil.WriteFile(filepath.Join(config.Bbsh, "cmd_"+config.CmdName+".go"), fullCode, 0444); err != nil {
+		if err := ioutil.WriteFile(filepath.Join(config.Bbsh, "cmd_"+c.CmdName+".go"), fullCode, 0444); err != nil {
 			log.Fatalf("%v\n", err)
 		}
 	}
 
 	return nil
 }
 
-func oneCmd() {
+func oneCmd(c Command) {
 	// Create the directory for the package.
 	// For now, ./cmds/<package name>
-	packageDir := filepath.Join(config.Bbsh, "cmds", config.CmdName)
+	packageDir := filepath.Join(config.Bbsh, "cmds", c.CmdName)
 	if err := os.MkdirAll(packageDir, 0755); err != nil {
 		log.Fatalf("Can't create target directory: %v", err)
 	}
 
 	fset := token.NewFileSet()
-	config.FullPath = filepath.Join(config.Gopath, config.CmdPath)
-	p, err := parser.ParseDir(fset, config.FullPath, nil, 0)
+	c.FullPath = filepath.Join(c.Gopath, c.CmdPath)
```

Comment:
It's kind of a convenience thing. I almost named this Urootgopath and maybe I should, the intent is that Gopath is where uroot commands are found.